### PR TITLE
DOCS-1533: Add build-action information

### PR DESCRIPTION
**Pull request by npentrel**

### CODE REVIEW

File: docs/registry/upload/_index.md [hunk 1]
```diff
@@ -288,47 +290,120 @@ To update an existing module in the [Viam registry](https://app.viam.com/registr
...
+  "module_id": "example-module",
+  ...
+  "build": {
+    "build": "make module.tar.gz",
+    "arch" : ["linux/amd64", "linux/arm64", "darwin/arm64"]
+  }
+}
+```
+
+You can test this command by running the following command on your development machine:
```
Comment (abe-winter) [inline docs/registry/upload/_index.md:334]: 'test this command' felt weird, might say 'config', up to you

**Review by abe-winter — APPROVED** [review]
!!

File: docs/registry/upload/_index.md (hunk 1 above)
Comment (npentrel) [inline docs/registry/upload/_index.md:334]:
```suggestion
You can test this build configuration by running the following command on your development machine:
```

**Comment by abe-winter** [comment]
ah also:
- important: tell users that this is only supported on public git repositories for now. (we can't clone private code)
- unless you are opposed, tell users cloud build is a beta product and may have some warts

File: docs/registry/upload/_index.md [hunk 2]
```diff
@@ -288,47 +290,120 @@ To update an existing module in the [Viam registry](https://app.viam.com/registr
...
-             key-id: ${{ secrets.viam_key_id }}
-             key-value: ${{ secrets.viam_key_value }}
-   ```
+1. Select the `build-action` or `upload-module` action and paste the following action template YAML into the edit window.
+
+   {{< tabs >}}
+   {{% tab name="CI with build-action" %}}
+
+```yaml {class="line-numbers linkable-line-numbers"}
+# see https://github.com/viamrobotics/upload-module for help
```
Comment (andf-viam) [inline docs/registry/upload/_index.md:299]:
```suggestion
# see https://github.com/viamrobotics/build-action for help
```

Think this is just an upstream typo, made: https://github.com/viamrobotics/build-action/pull/2

File: docs/registry/upload/_index.md [hunk 3]
```diff
@@ -288,47 +290,120 @@ To update an existing module in the [Viam registry](https://app.viam.com/registr
...
-         - uses: viamrobotics/upload-module@v1
-           # if: github.event_name == 'release' # <-- once the action is working, uncomment this so you only upload on release
-           with:
-             module-path: module.tar.gz
-             platform: linux/amd64 # <-- replace with your target architecture, or your module will not deploy
-             version: ${{ github.event_name == 'release' && github.ref_name || format('0.0.0-{0}.{1}', github.ref_name, github.run_number) }} # <-- see 'Versioning' section below for explanation
-             key-id: ${{ secrets.viam_key_id }}
-             key-value: ${{ secrets.viam_key_value }}
-   ```
+1. Select the `build-action` or `upload-module` action and paste the following action template YAML into the edit window.
```
Comment (andf-viam) [inline docs/registry/upload/_index.md:293]: Nit: `select` suggests a UI element in the GH UI that the user should find and interact with. We mean: pick the right Docs tab below. What do you think of:

```suggestion
1. Paste the following action template YAML into the edit window, depending on whether you are using the `build-action` or `upload-module` action:
```

File: docs/registry/upload/_index.md [hunk 4]
```diff
@@ -288,47 +290,120 @@ To update an existing module in the [Viam registry](https://app.viam.com/registr
...
+        with:
+          # note: you can replace this line with 'version: ""' if
+          # you want to test the build process without deploying
+          version: ${{ github.ref_name }}
+          ref: ${{ github.sha }}
+          key-id: ${{ secrets.viam_key_id }}
+          key-value: ${{ secrets.viam_key_value }}
+```
+
+The `build-action` relies on a build command that you need to specify in the <file>meta.json</file> file that you created for your module when you first [uploaded it](/registry/upload/#upload-a-custom-module).
```
Comment (andf-viam) [inline docs/registry/upload/_index.md:320]: Nit:

```suggestion
The `build-action` GitHub action relies on a build command that you need to specify in the <file>meta.json</file> file that you created for your module when you first [uploaded it](/registry/upload/#upload-a-custom-module).
```

Or

```suggestion
`build-action` relies on a build command that you need to specify in the <file>meta.json</file> file that you created for your module when you first [uploaded it](/registry/upload/#upload-a-custom-module).
```

File: docs/registry/upload/_index.md [hunk 5]
```diff
@@ -288,47 +290,120 @@ To update an existing module in the [Viam registry](https://app.viam.com/registr
...
+    "build": "make module.tar.gz",
+    "arch" : ["linux/amd64", "linux/arm64", "darwin/arm64"]
+  }
+}
+```
+
+You can test this build configuration by running the following command on your development machine:
+
+```sh {class="command-line" data-prompt="$"}
+viam module build local
```
Comment (andf-viam) [inline docs/registry/upload/_index.md:337]: @abe-winter do we need `viam module build start` for any part of using -- or verifying -- this GH action?

File: docs/registry/upload/_index.md [hunk 6]
```diff
@@ -288,47 +290,120 @@ To update an existing module in the [Viam registry](https://app.viam.com/registr
...
+```
+
+The command will run your build instructions without running a cloud build job.
```
Comment (andf-viam) [inline docs/registry/upload/_index.md:340]: Supernit: `local` for what it does do, not just `cloud` for what it doesn't. Even though it's in the command name:

```suggestion
The command will run your build instructions locally without running a cloud build job.
```

File: docs/registry/upload/_index.md [hunk 7]
```diff
@@ -288,47 +290,120 @@ To update an existing module in the [Viam registry](https://app.viam.com/registr
...
+          platform: linux/amd64 # <-- replace with your target architecture, or your module will not deploy
+          version: ${{ github.event_name == 'release' && github.ref_name || format('0.0.0-{0}.{1}', github.ref_name, github.run_number) }} # <-- see 'Versioning' section below for explanation
+          key-id: ${{ secrets.viam_key_id }}
+          key-value: ${{ secrets.viam_key_value }}
+```
+
+Edit the copied code to include the configuration specific to your module.
+Each item marked with a `<--` comment requires that you edit the configuration values accordingly.
+
+Set `run` to the command you use to build and package your module.
```
Comment (andf-viam) [inline docs/registry/upload/_index.md:378]: Opt: just offering a little more context:

```suggestion
Set `run` to the command you use to build and package your module, such as invoking a makefile or running a shell script.
```

**Review by andf-viam — APPROVED** [review]
LGTM, just some optional nits!

File: docs/registry/upload/_index.md (hunk 5 above)
Comment (abe-winter) [reply docs/registry/upload/_index.md:337]: using no, verifying maybe

you can run `viam module build start` on CLI if you want, and then `viam module build logs --id whatever`

(with the caveat that you may quickly hit your daily limit of 30 builds per org)

it makes the iteration loop slightly shorter if someone is debugging a failing build

File: docs/registry/upload/_index.md [hunk 8]
```diff
@@ -279,7 +279,9 @@ For more information, see the [`viam module` command](/fleet/cli/#module)
 
 ### Update an existing module using a GitHub action
 
-To update an existing module in the [Viam registry](https://app.viam.com/registry) using continuous integration (CI), use the [`upload-module` GitHub action](https://github.com/viamrobotics/upload-module).
+To update an existing module in the [Viam registry](https://app.viam.com/registry) using continuous integration (CI), you can use one of two Github actions.
```
Comment (abe-winter) [inline docs/registry/upload/_index.md:282]: > update an existing module

'existing module' part matters a lot

if `viam module create` and `viam module update` have not been run, cloud builds will fail (with useful error messages though)

your call whether to emphasize this here, or let users find out

File: docs/registry/upload/_index.md (hunk 5 above)
Comment (andf-viam) [reply docs/registry/upload/_index.md:337]: Gothca, I'd omit it from here then imo. Thanks for the annswer!

File: docs/registry/upload/_index.md [hunk 9]
```diff
@@ -288,47 +290,120 @@ To update an existing module in the [Viam registry](https://app.viam.com/registr
...
-         - uses: viamrobotics/upload-module@v1
-           # if: github.event_name == 'release' # <-- once the action is working, uncomment this so you only upload on release
-           with:
-             module-path: module.tar.gz
-             platform: linux/amd64 # <-- replace with your target architecture, or your module will not deploy
-             version: ${{ github.event_name == 'release' && github.ref_name || format('0.0.0-{0}.{1}', github.ref_name, github.run_number) }} # <-- see 'Versioning' section below for explanation
-             key-id: ${{ secrets.viam_key_id }}
-             key-value: ${{ secrets.viam_key_value }}
-   ```
+1. Paste the following action template YAML into the edit window, depending on whether you are using the `build-action` or `upload-module` action:
```
Comment (npentrel) [inline docs/registry/upload/_index.md:293]:
```suggestion
1. Paste one of the following action templates into the edit window, depending on whether you are using the `build-action` or `upload-module` action:
```

File: docs/registry/upload/_index.md (hunk 5 above)
Comment (npentrel) [reply docs/registry/upload/_index.md:337]: You mean omit the build start or the command we've put here?

File: docs/registry/upload/_index.md (hunk 8 above)
Comment (npentrel) [inline docs/registry/upload/_index.md:282]:
```suggestion
To update an existing module in the [Viam registry](https://app.viam.com/registry) using continuous integration (CI), you can use one of two Github actions.
You can only use these GitHub actions if you have already created the module by running `viam module create` and `viam module update`.
```

Comment (npentrel) [reply docs/registry/upload/_index.md:282]: Added

**Pull request merged**